docs/contributing/guidelines-for-pull-request.md

```diff
@@ -38,6 +34,15 @@ E.g., if the latest release is `v7.1.0` and you want to introduce a breaking cha
     * you do not have to update `CHANGELOG.md` at all as it is generated automatically using [symplify/changelog-linker](https://github.com/symplify/changeloglinker) during our release process.
 * Now just wait for review of your change.
 
+### Note about targeting pull requests
+As we [support multiple versions](./backward-compatibility-promise.md#current-release-plan) of Shopsys Framework while keeping [Backward Compatibility Promise](/docs/contributing/backward-compatibility-promise.md), it is important to think about proper targeting of your pull request.
+
+If your pull request:
+* **fixes a bug and does not contain any BC break**, it should be targeted to the oldest supported version where the bug occurs.
+* **does not contain any BC break**, it should be targeted to `master`.
+* **contains any BC break**, it should be targeted to a branch where the next major release is being prepared.
+    * E.g., if the latest release is `v7.1.0` and you want to introduce a breaking change, you need to rebase your branch on `8.0` branch and target your PR against it. If no such a branch exists, you need to create one.
+
```

Comment:
```suggestion

*Note: Always rebase your branch onto the base branch before retargeting, otherwise your PR might contain more commits than you'd want to merge.*

```